solidity/test/rewards/TestECDSARewards.js

```diff
@@ -268,11 +342,88 @@ describe("ECDSARewards", () => {
+      const keep0 = await BondedECDSAKeepStub.at(keepAddress0)
+      await keep0.publicMarkAsClosed()
+
+      // Full allocation for the first interval would be
+      // 178,200,000 * 4% = 7,128,000.
+      // Because just 1% of minimum keep quota is met, the allocation is
+      // 7,128,000 * 1% = 71,280.
+      // keeps created: 10 => 7,128 KEEP per keep
+      // member receives: 7,128 / 3 = 2,376 (3 signers per keep)
+      await rewardsContract.receiveReward(keepAddress0)
+      await withdrawRewards(0, operators)
+
+      let expectedBeneficiaryBalance = new BN(2376)
+      await assertKeepBalanceOfBeneficiaries(expectedBeneficiaryBalance)
+
+      // Second keep will be closed a day later
+      await time.increase(time.duration.days(1))
+
+      const keepAddress1 = await keepFactory.getKeepAtIndex(1)
+      const keep1 = await BondedECDSAKeepStub.at(keepAddress1)
+      await keep1.publicMarkAsClosed()
+
+      await rewardsContract.receiveReward(keepAddress1)
+      await withdrawRewards(0, operators)
+
+      // each member was in 2 properly closed keeps: 2,376 * 2 = 4,752
+      expectedBeneficiaryBalance = new BN(4752)
+      await assertKeepBalanceOfBeneficiaries(expectedBeneficiaryBalance)
+    })
+
+    it("should correctly track withdrawable and withdrawn amount", async () => {
+      for (let i = 0; i < 8; i++) {
+        await keepFactory.stubOpenKeep(owner, operators, firstIntervalStart)
+      }
+
+      await timeJumpToEndOfIntervalIfApplicable(0)
+
+      // mark first keep as closed
+      const keepAddress0 = await keepFactory.getKeepAtIndex(0)
+      const keep0 = await BondedECDSAKeepStub.at(keepAddress0)
+      await keep0.publicMarkAsClosed()
+
+      // mark second keep as closed
+      const keepAddress1 = await keepFactory.getKeepAtIndex(1)
+      const keep1 = await BondedECDSAKeepStub.at(keepAddress1)
+      await keep1.publicMarkAsClosed()
+
+      await rewardsContract.receiveReward(keepAddress0)
+      await rewardsContract.receiveReward(keepAddress1)
+
+      // Full allocation for the first interval would be
+      // 178,200,000 * 4% = 7,128,000.
+      // Because just 0.8% of minimum keep quota is met, the allocation is
+      // 7,128,000 * 0.8% = 57,024.
+      // keeps created: 8 => 7,128 KEEP per keep
+      // member receives: 7,128 / 3 = 2,376 (3 signers per keep)
+      const expectedBeneficiaryBalance = new BN(2376)
+      // 2,376 * 2 = 4,752
+      const expectedBalanceFromTwoKeeps = expectedBeneficiaryBalance.muln(2)
+
+      await assertWithdrawableRewards(expectedBalanceFromTwoKeeps, 0, operators)
+      await assertWithdrawnRewards(new BN(0), 0, operators)
+
+      await withdrawRewards(0, operators)
+
+      await assertWithdrawableRewards(new BN(0), 0, operators)
+      await assertWithdrawnRewards(expectedBalanceFromTwoKeeps, 0, operators)
+    })
   })
 
   async function assertKeepBalanceOfBeneficiaries(expectedBalance) {
```

Comment:
Can we add `operators` as a parameter here? Just to stay consistent with the rest of `assert*` functions.

Reply:
Hm, `operators` are actually visible by all the functions in tests which is set in `before`. Maybe let's do the other way around and remove this parameter from `assert*` functions?

Reply:
Yeah, we can do that.